### DISCUSSION TITLE
Re-add support for declaring data source URL in config file

### DISCUSSION THREAD

**Discussion by tothbotond** [grafana/grafana-infinity-datasource#367] [Ideas]
Hi,

We are using your Grafana plugin to source data for our dashboards. As we are managing **multiple** Grafana instances, we would like the option to support loading the data source url config property via environment variables, like this:

_in our datasources.yaml:_
![Screenshot 2022-07-25 at 10 03 33](https://user-images.githubusercontent.com/108657045/180728753-f79f21be-f66b-4e45-a63d-47a2551ab2f6.png)

However, according to your plugin history, the datasource URL field is deprecated and ignored from the configuration:
![Screenshot 2022-07-25 at 10 05 54](https://user-images.githubusercontent.com/108657045/180729022-dfc28bf7-8803-4da9-abaa-ea3fc411fc1f.png)

The suggestion, to use the query editor's URL field is not applicable to us, as Grafana does not support resolving environment variables for that field, and it will not be supported according to their documentation.

The field was made deprecated in this commit: https://github.com/yesoreyeram/grafana-infinity-datasource/issues/120

Would it be possible to readd the URL config property (or do you have any idea how could we achieve this)? If this is something where we could help as contributors, that would also be great for us.

Thanks for your help in advance,
Botond

**Comment by tothbotond** [comment]
@yesoreyeram Hi, could you please give your feedback on this?

**Reply by yesoreyeram** [reply]
Hi.. Sorry missed the post. Any specific reason why you want this field in settings? 

If you want to change URL by environment, Can't you achieve the same via variables? I am looking for strong usecases why this will be helpful.

**Reply by tothbotond** [reply]
Hi, no worries, thanks for getting back to us.

We are managing self hosted Grafana instances with an IaC tool (Cloudformation) on AWS ECS. Having the ability to use env variables to specify those base URLs for each of our Grafana instances (dev/test/staging/prod etc.) would be very convinient for us with our current DevOps setup.

**Reply by bukowa** [reply]
> Hi.. Sorry missed the post. Any specific reason why you want this field in settings?
> 
> If you want to change URL by environment, Can't you achieve the same via variables? I am looking for strong usecases why this will be helpful.

Imagine `variable` that is based on data source:
![image](https://user-images.githubusercontent.com/38087302/231874544-e4a626fd-f85e-41e7-ab3c-e6a60b84feee.png)

Now for each datasource i want to duplicate panels:
![image](https://user-images.githubusercontent.com/38087302/231874650-3f6a90eb-7784-4db2-a077-3c9e66631b76.png)

So i have to input URL twice:
![image](https://user-images.githubusercontent.com/38087302/231874716-3c7b8e11-d9e2-4981-b954-f0b42b8007d3.png)

And each URL has different auth:
![image](https://user-images.githubusercontent.com/38087302/231874803-2d45730c-3658-49f8-86ed-001b765afacb.png)

And may have different allowed hosts:
![image](https://user-images.githubusercontent.com/38087302/231874894-3e91c94b-00b2-47b5-97db-a58ed0f5a0b9.png)

So i would imagine to just specify path without url:
![image](https://user-images.githubusercontent.com/38087302/231874999-7c65f7e9-155c-4406-870c-8fd5b44be54c.png)

And then `infinity` would resolve to `BASE_URL` from data source configuration?

**Comment by kevinlinxc** [comment]
If I used a variable, I would need that variable for every dashboard, right? It seems like just having the URL option in the settings would be much more convenient, especially because people making new dashboards with my setup might not know about the variables. I also have the same use case of having a different URL depending on environment (local/dev/prod)

**Comment by kevinlinxc** [comment]
@yesoreyeram can we expect URLs to be brought back?

**Reply by yesoreyeram** [reply]
URL won't be going away. but That's not preferred way. Use full URL in the query editor with allowed URL list in the config.

**Comment by mykael22000** [comment]
I too, would find a properly supported base_url very useful.  Some instructions on how to use it in a panel would also be appreciated.

We ship a product that stores its data in a database. Each customer gets to set up their own database, with its own URL.

We are considering shipping a number of Grafana dashboards for our customers, provided we can make the installation simple enough.  Each dashboard can contain as many as a dozen infinity queries (variables and panels) and we may have a couple of dozen dashboards.

Ideally the installation process would be along the lines of:

1. Customer creates an infinity datasource and sets its base_url and allowed_urls to point to their database.
2. Customer imports the dashboards and links them to their infinity datasource - and they just work.

Having to edit each dashboard to set a base_url variable (or, worse, each query to change the URL) is far from optimal.  It also spoils their first look at our dashboards as they will come up with no data and/or errors because the URL for the queries is incorrect.   This also means that we can ship them updates to the dashboard that will not require an arduous re-customization exercise on their part once they install them.  Ideally the customer should never have to edit our dashboards.
